keps/sig-network/3178-iptables-cleanup/README.md

```diff
@@ -0,0 +1,764 @@
+startup, so for the purposes of this KEP there is no real difference
+between starting a fresh kube-proxy and upgrading an existing one.
+
+### Version Skew Strategy
+
+As long as we wait two releases between Alpha and Beta, then all
+allowed skewed versions of kubelet and kube-proxy will be compatible
+with each other.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in valuesin `kep.yaml`)
+  - Feature gate name: IPTablesCleanup
+  - Components depending on the feature gate:
+    - kubelet
+
+###### Does enabling the feature change any default behavior?
+
+When the feature gate is enabled, kubelet will no longer create the
+IPTables chains/rules that it used to. This may cause problems with
+third-party components in Alpha but these problems are expected to be
+ironed out before moving to Beta.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing unexpected
+
+###### Are there any tests for feature enablement/disablement?
+
+No... there is no real difference between enabling the feature in an
+existing cluster vs creating a cluster where it was always enabled.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The most likely cause of a rollout failure would be a third-party
+component that depended on one of the no-longer-existing IPTables
+chains. It is impossible to predict exactly how this third-party
+component would fail in this case, but it would likely impact already
+running workloads.
+
+###### What specific metrics should inform a rollback?
+
+Any failures would be the result of third-party components being
+incompatible with the change, so no core Kubernetes metrics are likely
+to be relevant.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+TBD
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+This KEP will eventually remove two kubelet command-line arguments,
+but not until after the feature is GA.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+There is no simple way to do this because if the feature is working
+correctly there will be no difference in externally-visible behavior.
+(The generated iptables rules will be different, but the _effect_ of
+the generated iptables rules will be the same.)
```

Comment:
This feature isn't really "used by workloads" so much as it is "configured by the administrator", so you can say something that effect.